examples/azure-todo/rg.yaml

```diff
@@ -0,0 +1,147 @@
+    kind: AzureToDoDeployment
+    spec:
+      name: string | default=my-todo
+      namespace: string | default=default
+      location: string | required=true
+  resources:
+  # These are the resources in Azure needed to run the TODO app
+  - id: rg
+    template:
+      apiVersion: resources.azure.com/v1api20200601
+      kind: ResourceGroup
+      metadata:
+        name: ${schema.spec.name}-rg
+        namespace: ${schema.spec.namespace}
+      spec:
+        location: ${schema.spec.location}
+  - id: databaseAccount
+    template:
+      apiVersion: documentdb.azure.com/v1api20210515
+      kind: DatabaseAccount
+      metadata:
+        name: ${schema.spec.name}-sqldb
+        namespace: ${schema.spec.namespace}
+      spec:
+        location: ${schema.spec.location}
+        owner:
+          name: ${rg.spec.azureName}  # TODO: Fix this to be resourceGroup
+        kind: GlobalDocumentDB
+        databaseAccountOfferType: Standard
+        locations:
+          - locationName: ${schema.spec.location}
+        operatorSpec:
+          secrets:
+            primaryMasterKey:
+              name: ${schema.spec.name}-sqldb-connection-settings
+              key: key
+            documentEndpoint:
+              name: ${schema.spec.name}-sqldb-connection-settings
+              key: documentEndpoint
+  - id: database
+    template:
+      apiVersion: documentdb.azure.com/v1api20210515
+      kind: SqlDatabase
+      metadata:
+        name: ${schema.spec.name}-sql-db
+        namespace: ${schema.spec.namespace}
+      spec:
+        location: ${schema.spec.location}
+        owner:
+          name: ${databaseAccount.spec.azureName}
+        options:
+          autoscaleSettings:
+            maxThroughput: 4000
+        resource:
+          id: ${schema.spec.name}-sql-db
+  - id: container
+    template:
+      apiVersion: documentdb.azure.com/v1api20210515
+      kind: SqlDatabaseContainer
+      metadata:
+        name: ${schema.spec.name}-sql-container
+        namespace: ${schema.spec.namespace}
+      spec:
+        location: ${schema.spec.location}
+        owner:
+          name: ${database.spec.azureName}
+        resource:
+          id: ${schema.spec.name}-sql-container
+          partitionKey:
+            kind: Hash
+            paths: ["/id"]
+  # These are the resources of the TODO app
+  - id: app
+    template:
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        labels:
+          app: ${schema.spec.name}
+        name: ${schema.spec.name}
+        namespace: ${schema.spec.namespace}
+      spec:
+        replicas: 1
+        selector:
+          matchLabels:
+            app: ${schema.spec.name}
+        template:
+          metadata:
+            labels:
+              app: ${schema.spec.name}
+          spec:
+            containers:
+            - name: app
+              image: mcr.microsoft.com/k8s/asodemos/cosmostodo:latest
+              env:
+                - name: CosmosDB__Account
```

Comment:
Why the double underscore?

Reply:
It's just the format of the env that the cosmostodo container accepts - this was adapted from an existing Cosmos sample so we didn't pick the env format. I agree it's a bit weird =)